src/Hosting.CommandLine/CliArgs.cs

```diff
@@ -0,0 +1,12 @@
+namespace McMaster.Extensions.Hosting.CommandLine
+{
+    /// <summary>
+    /// A DI container for storing command line arguments.
+    /// </summary>
+    /// <seealso cref="HostBuilderExtensions.UseCli{T}(Microsoft.Extensions.Hosting.IHostBuilder, string[])"/>
+    public class CliArgs
```

Comment:
I wonder if this type is actually necessary. I think we could use `CommandLineContext` instead.

https://github.com/natemcmaster/CommandLineUtils/blob/master/src/CommandLineUtils/Abstractions/CommandLineContext.cs

Reply:
Perhaps...  but didn't work for me.  After modifying `CliService`:
```
        public CliService(ILogger<CliService<T>> logger, CommandLineContext context,
            IServiceProvider serviceProvider)
```
I end up with an `InvalidOperationException`:
```
Unhandled Exception: System.InvalidOperationException: Unable to resolve service for type 'McMaster.Extensions.CommandLineUtils.Abstractions.CommandLineContext' while attempting to activate 'McMaster.Extensions.Hosting.CommandLine.Internal.CliService`1[CustomServices.Program]'.
```
The [docs](https://github.com/natemcmaster/CommandLineUtils/blob/master/docs/docs/advanced/dependency-injection.md) say it should be available, but it doesn't seem to be.  I'm thinking it may be a 2 phased container initialization...  The services configured in the generic host are needed before the command line application starts at which point it adds its services?  But dunno...  You have an idea here?

Reply:
Joining service providers is something I've seen a few other users try to do. It might be worth doing, but we can treat that as a separate issue. Thanks for looking into this more.

Reply:
Ok, we will leave it for another issue.